src/spec-node/featuresCLI/upgrade.ts

```diff
@@ -0,0 +1,128 @@
+			'log-level': { choices: ['error' as 'error', 'info' as 'info', 'debug' as 'debug', 'trace' as 'trace'], default: 'info' as 'info', description: 'Log level.' },
+			'init-lockfile': { type: 'boolean', description: 'Initialize lockfile if it does not exist.', default: false },
+		});
+}
+
+export type FeaturesUpgradeArgs = UnpackArgv<ReturnType<typeof featuresUpgradeOptions>>;
+
+export function featuresUpgradeHandler(args: FeaturesUpgradeArgs) {
+	(async () => await featuresUpgrade(args))().catch(console.error);
+}
+
+async function featuresUpgrade({
+	'workspace-folder': workspaceFolderArg,
+	'docker-path': dockerPath,
+	config: configArg,
+	'docker-compose-path': dockerComposePath,
+	'log-level': inputLogLevel,
+	'init-lockfile': initLockfile,
+}: FeaturesUpgradeArgs) {
+	const disposables: (() => Promise<unknown> | undefined)[] = [];
+	const dispose = async () => {
+		await Promise.all(disposables.map(d => d()));
+	};
+	let output: Log | undefined;
+	try {
+		const workspaceFolder = path.resolve(process.cwd(), workspaceFolderArg);
+		const configFile = configArg ? URI.file(path.resolve(process.cwd(), configArg)) : undefined;
+		const cliHost = await getCLIHost(workspaceFolder, loadNativeModule, true);
+		const extensionPath = path.join(__dirname, '..', '..');
+		const sessionStart = new Date();
+		const pkg = getPackageConfig();
+		const output = createLog({
+			logLevel: mapLogLevel(inputLogLevel),
+			logFormat: 'text',
+			log: text => process.stderr.write(text),
+			terminalDimensions: undefined,
+		}, pkg, sessionStart, disposables);
+		const dockerComposeCLI = dockerComposeCLIConfig({
+			exec: cliHost.exec,
+			env: cliHost.env,
+			output,
+		}, dockerPath, dockerComposePath);
+		const dockerParams: DockerCLIParameters = {
+			cliHost,
+			dockerCLI: dockerPath,
+			dockerComposeCLI,
+			env: cliHost.env,
+			output,
+		};
+
+		const workspace = workspaceFromPath(cliHost.path, workspaceFolder);
+		const configPath = configFile ? configFile : await getDevContainerConfigPathIn(cliHost, workspace.configFolderPath);
+		const configs = configPath && await readDevContainerConfigFile(cliHost, workspace, configPath, true, output) || undefined;
+		if (!configs) {
+			throw new ContainerError({ description: `Dev container config (${uriToFsPath(configFile || getDefaultDevContainerConfigPath(cliHost, workspace!.configFolderPath), cliHost.platform)}) not found.` });
+		}
+		const config = configs.config.config;
+		const cacheFolder = await getCacheFolder(cliHost);
+		const params = {
+			extensionPath,
+			cacheFolder,
+			cwd: cliHost.cwd,
+			output,
+			env: cliHost.env,
+			skipFeatureAutoMapping: false,
+			platform: cliHost.platform,
+		};
+
+		const lockfilePath = getLockfilePath(config);
+
+		// Check if lockfile exists
+		if (!initLockfile && !(await cliHost.isFile(lockfilePath))) {
```

Comment:
Or just write the lockfile when it does not exist yet? That way we don't need the `--init-lockfile` option.